### DISCUSSION TITLE
Multi-nodes recognized as one node

### DISCUSSION THREAD

**Discussion by takahh** [horovod/horovod#3484] [Q&A]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) TF
2. Framework version: 2.4.0
3. Horovod version: 0.24.2
4. MPI version:3.1.4
5. CUDA version:11.2.146
6. NCCL version:2.8.4
7. Python version:3.8.3
8. Spark / PySpark version:
9. Ray version:
10. OS and version:Linux SUSE
11. GCC version:8.3.0
12. CMake version:3.22.2

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)? 
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.
[tape_aug_full.txt](https://github.com/horovod/horovod/files/8311164/tape_aug_full.txt)

Installation was ok but when I ran the horovodrun with two nodes (one GPU for each), 2GPUs in total.
The two GPU were recognized as two GPUs in the same node and I got OOM. The output is attached. Thank you.

#######
Horovod v0.24.2:

Available Frameworks:
    [X] TensorFlow
    [X] PyTorch
    [ ] MXNet

Available Controllers:
    [X] MPI
    [ ] Gloo

Available Tensor Operations:
    [X] NCCL
    [ ] DDL
    [ ] CCL
    [X] MPI
    [ ] Gloo

**Comment by takahh** [comment]
[test_tape_aug.txt](https://github.com/horovod/horovod/files/8311658/test_tape_aug.txt)

**Comment by takahh** [comment]
is the shell script I used. Thank you.

**Comment by takahh** [comment]
I used openmpi 2.1.2 and it didnt work either. openmpi 4.x was not available in the HPC.

**Comment by takahh** [comment]
I reinstalled horovod with openmpi 4.1.4 and got 
<pre><code>
[0] 2022-03-21 04:10:45.473329: I tensorflow/core/common_runtime/gpu/http://gpu_device.cc:1871] Adding visible gpu devices: 0
[1] 2022-03-21 04:10:45.473331: I tensorflow/core/common_runtime/gpu/http://gpu_device.cc:1871] Adding visible gpu devices: 0
[1] Traceback (most recent call last):
[1]  File “/gs/hs0/tga-science/kimura/transformer_tape_dnabert/python/multi_nodes/multi_nodes.py”, line 36, in <module>
[1]   tf.config.experimental.set_visible_devices(gpus[hvd.local_rank()], ‘GPU’)
[1] IndexError: list index out of range
</code></pre>

**Comment by takahh** [comment]
[tape_aug_full.txt](https://github.com/horovod/horovod/files/8311999/tape_aug_full.txt)
the full stderr atached

**Comment by takahh** [comment]
[script.txt](https://github.com/horovod/horovod/files/8312000/script.txt)
Shell script I used attached

**Comment by takahh** [comment]
<pre><code>
print(' gpus = ', gpus)
print(socket.gethostname())
print(f"hvd.rank() {hvd.rank()}")
print(f"hvd.local_rank() {hvd.local_rank()}")

[1]  gpus =  [PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
[1] r2i1n8
[1] hvd.rank() 1
[1] hvd.local_rank() 1
[0]  gpus =  [PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
[0] r2i1n8
[0] hvd.rank() 0
[0] hvd.local_rank() 0
</pre></code>
Looks like horovod sees just one node and assign two?

**Comment by maxhgerlach** [comment]
You will need to tell `horovodrun` about the hosts that you want to use. Check out the "host arguments" section in `horovodrun --help`. 

It looks like you work in an HPC cluster environment. Advise will be quite specific to such a system.

**Comment by takahh** [comment]
Hi, @maxhgerlach. Yes. I am using HPC. Thank you for the advice. I was not sure about it. I will close this and give it a try.
